packages/ember-testing/tests/adapters/qunit_test.js

```diff
@@ -1,51 +1,55 @@
 import { run } from 'ember-metal';
 import QUnitAdapter from '../../adapters/qunit';
+import { moduleFor, AbstractTestCase } from 'internal-test-helpers';
 
 var adapter;
 
-QUnit.module('ember-testing QUnitAdapter', {
-  setup() {
+moduleFor('ember-testing QUnitAdapter', class extends AbstractTestCase {
+  constructor() {
+    super();
     adapter = new QUnitAdapter();
-  },
+  }
+
   teardown() {
     run(adapter, adapter.destroy);
   }
-});
 
-QUnit.test('asyncStart calls stop', function() {
-  var originalStop = QUnit.stop;
-  try {
-    QUnit.stop = function() {
-      ok(true, 'stop called');
-    };
-    adapter.asyncStart();
-  } finally {
-    QUnit.stop = originalStop;
+  ['@test asyncStart calls stop'](assert) {
+    var originalStop = QUnit.stop;
+    try {
+      QUnit.stop = function() {
+        assert.ok(true, 'stop called');
+      };
+      adapter.asyncStart();
+    } finally {
+      QUnit.stop = originalStop;
+    }
   }
-});
 
-QUnit.test('asyncEnd calls start', function() {
-  var originalStart = QUnit.start;
-  try {
-    QUnit.start = function() {
-      ok(true, 'start called');
-    };
-    adapter.asyncEnd();
-  } finally {
-    QUnit.start = originalStart;
+  ['@test asyncEnd calls start'](assert) {
+    var originalStart = QUnit.start;
+    try {
+      QUnit.start = function() {
+        assert.ok(true, 'start called');
+      };
+      adapter.asyncEnd();
+    } finally {
+      QUnit.start = originalStart;
+    }
   }
-});
 
-QUnit.test('exception causes a failing assertion', function() {
-  var error = { err: 'hai' };
-  var originalOk = window.ok;
-  try {
-    window.ok = function(val, msg) {
-      originalOk(!val, 'ok is called with false');
-      originalOk(msg, '{err: "hai"}');
-    };
-    adapter.exception(error);
-  } finally {
-    window.ok = originalOk;
+  ['@test exception causes a failing assertion']() {
+    var error = { err: 'hai' };
+    var originalOk = window.ok;
+    try {
+      window.ok = function(val, msg) {
```

Comment:
This is also unclear.

Reply:
I have some tests for this in ember-qunit, basically you should do:

```js
    let originalPushResult = assert.pushResult;
    assert.pushResult = function(resultInfo) {
      // Inverts the result so we can test failing assertions
      resultInfo.result = !resultInfo.result;
      resultInfo.message = `Failed: ${resultInfo.message}`;
      originalPushResult(resultInfo);
    };
```